### PR TITLE
pkp/pkp-lib#1131 Add CSRF checking

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: classes/controllers/grid/filter/PKPFilterGridHandler.inc.php [hunk 1]
```diff
@@ -258,9 +258,7 @@ function deleteFilter(&$args, $request) {
 		$filter = $this->getFilterFromArgs($request, $args);
 
 		$filterDao = DAORegistry::getDAO('FilterDAO');
-		$result = $filterDao->deleteObject($filter);
-
-		if ($result) {
+		if ($request->checkCSRF() && !$filterDao->deleteObject($filter)) {
```
Comment (bozana) [inline classes/controllers/grid/filter/PKPFilterGridHandler.inc.php]: should the negation "!" be there?

File: controllers/grid/admin/languages/PKPAdminLanguageGridHandler.inc.php [hunk 2]
```diff
@@ -330,30 +329,31 @@ function enableLocale($args, $request) {
...
 				);
-			} else {
-				$locale = $rowId;
-				$this->_updateLocaleSupportState($request, $rowId, false);
-
-				$notificationManager->createTrivialNotification(
-					$user->getId(), NOTIFICATION_TYPE_SUCCESS,
-					array('contents' => __('notification.localeDisabled'))
-				);
+				return new JSONMessage(false);
```
Comment (bozana) [inline controllers/grid/admin/languages/PKPAdminLanguageGridHandler.inc.php]: In order for the notification ('admin.languages.cantDisable') to be displayed this should be called here, I believe: `return DAO::getDataChangedEvent($locale);` -- as it was earlier. Else the confirmation modal just stays open and nothing happens...

File: controllers/grid/announcements/ManageAnnouncementGridHandler.inc.php [hunk 3]
```diff
@@ -122,13 +122,8 @@ function addAnnouncement($args, $request) {
 	 * @return JSONMessage JSON object
 	 */
 	function editAnnouncement($args, $request) {
-		$announcementId = (int)$request->getUserVar('announcementId');
-		$context = $request->getContext();
-		$contextId = $context->getId();
-
-		$announcementForm = new AnnouncementForm($contextId, $announcementId);
+		$announcementForm = new AnnouncementForm($context->getId(), (int) $request->getUserVar('announcementId'));
```
Comment (bozana) [inline controllers/grid/announcements/ManageAnnouncementGridHandler.inc.php]: Where does the $context come from?

File: classes/controllers/grid/filter/PKPFilterGridHandler.inc.php (hunk 1 above)
Comment (asmecher) [inline classes/controllers/grid/filter/PKPFilterGridHandler.inc.php]: Fixed, good catch!

File: controllers/grid/admin/languages/PKPAdminLanguageGridHandler.inc.php (hunk 2 above)
Comment (asmecher) [inline controllers/grid/admin/languages/PKPAdminLanguageGridHandler.inc.php]: Ah, yes, that makes sense. Fixed.

File: controllers/grid/announcements/ManageAnnouncementGridHandler.inc.php (hunk 3 above)
Comment (asmecher) [inline controllers/grid/announcements/ManageAnnouncementGridHandler.inc.php]: Already fixed: https://github.com/pkp/pkp-lib/commit/f59e508f91082fe50d7543c0765aa61c060aaff0

**Pull request merged**